### PR TITLE
Make the cmake & CI builds use the same settings

### LINKED ISSUE #72: Appveyor build inconsistent with cmake
The appveyor build adds compiler flags to the cmake build. Can these go into the cmake file, so the local & CI builds produce the same result?

Ubuntu (`clang`):
```
      CFLAGS: "-Wall -Wextra -pedantic -Werror -O1 -g -fsanitize=address -fsanitize=undefined -fno-sanitize-recover=all -fno-omit-frame-pointer"
      CXXFLAGS: "-Wall -Wextra -pedantic -Werror -O1 -g -fsanitize=address -fsanitize=undefined -fno-sanitize-recover=all -fno-omit-frame-pointer"

```

VS 2017 & 2019:
```
      CFLAGS: "/W3 /WX /DLIBDIVIDE_ASSERTIONS_ON"
      CXXFLAGS: "/W3 /WX /DLIBDIVIDE_ASSERTIONS_ON"
```

### PR DESCRIPTION
**Pull request by adbancroft**
1. Update the cmake build so it has the same compiler flags & options as the CI build
2. Change the CI build to use cmake only - no CI only options!

Fixes #72

### CODE REVIEW

**Comment by adbancroft** [comment]
I wonder if we'd be better off with a separate cmake build type? Rather than a flag that affects the existing build types.

File: README.md [hunk 1]
```diff
@@ -226,9 +226,14 @@ so benchmark is valuable for its verification as well.
...
-We currently do not have automated testing! Hence, before sending in patches, it would be nice
-if you compiled your new code at high warning levels using at least MSVC and GCC (or Clang).
-Also run the tester program to verify correctness and the benchmark programs to ensure you have
-not introduced any performance regressions.
+Although there are no individual unit tests, the supplied ```cmake``` builds do include several safety nets:
+
+* They compile with:
+  * All warnings on and;
+  * Warnings as errors
+* The CI build will build and run with sanitizers on ; these are available as part of the cmake build: ```-DSANITIZE:STRING=On```
```
Comment (ridiculousfish) [inline README.md:234]: This comment looks stale now, it should be `-DCMAKE_BUILD_TYPE=Sanitize`

**Review by ridiculousfish — APPROVED** [review]
Please squash commits as there's some churn around how to build with sanitizers.

On my machine, a test with `-DCMAKE_BUILD_TYPE=Sanitize` takes a very long time, mainly because of `benchmark_branchfree`. I think this will blow through any CI allocations. Let's not run `benchmark_branchfree` in CI.

**Comment by adbancroft** [comment]
> On my machine, a test with `-DCMAKE_BUILD_TYPE=Sanitize` takes a very long time, mainly because of `benchmark_branchfree`. I think this will blow through any CI allocations. Let's not run `benchmark_branchfree` in CI.

Appveyor allows 60 minutes _per job_. With a sanitize `benchmark_branchfree`, the max job time is ~11 mins (which includes the AVR sim run). It's adding ~3 mins to 2 of the builds, but not close to the limit.

Still want it removed?

**Comment by ridiculousfish** [comment]
> Appveyor allows 60 minutes _per job_. With a sanitize `benchmark_branchfree`, the max job time is ~11 mins (which includes the AVR sim run). It's adding ~3 mins to 2 of the builds, but not close to the limit.

Wow. Ok, let's keep it!

**Pull request merged**